test/metabase/api/field_test.clj

```diff
@@ -742,70 +742,81 @@
                                     [3 "Synergistic Granite Chair"]]}
                           (mt/user-http-request :crowberto :get 200 (format "field/%d/values" (mt/id :orders :product_id)))))))))))
 
-(deftest json-unfolding-default-true-test
+(deftest json-unfolding-initially-true-test
   (mt/test-drivers (mt/normal-drivers-with-feature :nested-field-columns)
     (when-not (mysql-test/is-mariadb? (u/id (mt/db)))
       (mt/dataset json
-        ;; Create a new database with the same details as the json dataset, with json unfolding enabled by default
+        ;; Create a new database with the same details as the json dataset, with json unfolding enabled
         (let [database (t2/select-one Database :id (mt/id))]
           (mt/with-temp* [Database [database {:engine driver/*driver*, :details (assoc (:details database) :json-unfolding true)}]]
             (mt/with-db database
               ;; Sync the new database
               (sync/sync-database! database)
               (let [field (t2/select-one Field :id (mt/id :json :json_bit))
-                    set-json-unfolding! (fn [v]
-                                          (mt/user-http-request :crowberto :put 200 (format "field/%d" (mt/id :json :json_bit))
-                                                                (assoc field :json_unfolding v)))
+                    get-field (fn [] (t2/select-one Field :id (mt/id :json :json_bit)))
+                    get-database (fn [] (t2/select-one Database :id (mt/id)))
+                    set-json-unfolding-for-field! (fn [v]
+                                                    (mt/user-http-request :crowberto :put 200 (format "field/%d" (mt/id :json :json_bit))
+                                                                          (assoc field :json_unfolding v)))
+                    set-json-unfolding-for-db! (fn [v]
+                                                 (let [updated-db (into {} (assoc-in database [:details :json-unfolding] v))]
+                                                   (mt/user-http-request :crowberto :put 200 (format "database/%d" (:id database))
+                                                                         updated-db)))
                     nested-fields          (fn []
                                              (->> (t2/select Field :table_id (mt/id :json) :active true :nfc_path [:not= nil])
                                                   (filter (fn [field] (= (first (:nfc_path field)) "json_bit")))))]
-                (testing "json-unfolding is enabled by default"
+                (testing "json_unfolding is enabled by default at the field level"
```

Comment:
nit: for readability I'll put a check that the json unfolding for db is enabled here.

Reply:
does that aid readability? To me that just adds more code to read. Line 751 clearly sets json-unfolding for the DB.